tests/polkadot_tests/arrays.rs

```diff
@@ -1413,6 +1414,26 @@ fn abi_encode_dynamic_array4() {
     );
 }
 
+#[test]
+fn abi_encode_dynamic_array5() {
+    let uint_str = "201f1e1d1c1b1a191817161514131211100f0e0d0c0b0a090807060504030201";
+    let uint_val = U256::from_str_radix(uint_str, 16).unwrap();
+
+    let mut runtime = build_solidity(&format!(
+        "contract Test {{
+            function test() external view returns (uint256[] memory _amounts) {{
+                _amounts = new uint256[](2);
+                _amounts[0] = 0x{};
+                _amounts[1] = 0x{};
+            }}
+        }}",
+        uint_str, uint_str
```

Comment:
I think a slightly better test would have two different values - or even, two random values.

Reply:
Please no random values, they make debugging failing tests painful :monkey: 

But I agree about using two different values, gonna change it before merging :+1:

Reply:
I think using random values is great, it gives much better coverage. You might get random failures but those identify real issues.

Reply:
Ok, so TIL about [monkey testing](https://en.wikipedia.org/wiki/Monkey_testing) actually being a thing.

Inspired by this I wrote another test, which monkey tests (using random values each time) the encoding and decoding of the array for _every_ array length between 0 and 128.